### PR TITLE
high-scores: Add test cases for array mutation

### PR DESCRIPTION
**Pull request by slaymance**
This is a PR in response to [Issue 931](https://github.com/exercism/javascript/issues/931) on the JavaScript track.

This exercise presents a problem in the JavaScript track due to array mutability (I know, classic). Many learners' implementations make use of the `Array.prototype.sort` method to get the top three personal scores. However, in JavaScript, this method mutates the array. This causes subsequent calls to retrieve the array of scores, as well as getting the latest score if the implementation is getting the last score from the scores array, to be different that if these methods were called before the sort occurred.

In JavaScript, our plan is to have some additional test cases that would check the `scores` array and `latestScore` property before and after a call to `personalTopThree` to ensure they are the same. We felt this would be a useful PR to the canonical tests since other languages may also have this issue.

I've placed the additional tests in the `Top 3 scores` block, but would like some input if they make the most sense there. I also created new properties to capture these test cases, though perhaps we may find that unnecessary if we can implement them using existing properties.

### CODE REVIEW

File: exercises/high-scores/canonical-data.json [hunk 1]
```diff
@@ -83,6 +83,24 @@
             "scores": [40]
           },
           "expected": [40]
+        },
+        {
+          "uuid": "",
```
Comment (SleeplessByte) [inline exercises/high-scores/canonical-data.json:88]:
```suggestion
          "uuid": "2df075f9-fec9-4756-8f40-98c52a11504f",
```

File: exercises/high-scores/canonical-data.json [hunk 2]
```diff
@@ -83,6 +83,24 @@
...
+          "description": "Latest score after personal top scores",
+          "property": "latestAfterTopThree",
+          "input": {
+            "scores": [70, 50, 20, 30]
+          },
+          "expected": 30
+        },
+        {
+          "uuid": "",
```
Comment (SleeplessByte) [inline exercises/high-scores/canonical-data.json:97]:
```suggestion
          "uuid": "809c4058-7eb1-4206-b01e-79238b9b71bc",
```

**Review by SleeplessByte — CHANGES_REQUESTED** [review]

**Comment by SleeplessByte** [comment]
I generated two UUID v4s for ya.

**Comment by SleeplessByte** [comment]
If this is merged after #1740, this PR should be updated to use the `immutable` scenario.
@exercism/reviewers want to check both out?

**Review by SleeplessByte — APPROVED** [review]

**Review by BethanyG — APPROVED** [review]

**Comment by glennj** [comment]
The Groovy track also thought about this: https://github.com/exercism/groovy/blob/master/exercises/high-scores/src/test/groovy/HighScoresSpec.groovy#L85

**Review by neenjaw — APPROVED** [review]

**Review by SleeplessByte — CHANGES_REQUESTED** [review]
Can you update both with the `immutable` scenario?

File: exercises/high-scores/canonical-data.json [hunk 3]
```diff
@@ -83,6 +83,30 @@
             "scores": [40]
           },
           "expected": [40]
+        },
+        {
+          "uuid": "2df075f9-fec9-4756-8f40-98c52a11504f",
+          "description": "Latest score after personal top scores",
+          "property": "latestAfterTopThree",
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply exercises/high-scores/canonical-data.json:90]: Too much meaning :). A lot of the exercises can be tested using property-based tests, but sometimes, like right now, we can't. In other exercises we've chosen to _abuse_ the property key adding new "properties", like this one, that name what they intend to do.

In most languages you _could_ implement them as a property on the student's solution, but moreoften you'll write it as a few steps to take inside the test. So in this case: call top three first, then test latest.

**Review by SleeplessByte — APPROVED** [review]

File: exercises/high-scores/canonical-data.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply exercises/high-scores/canonical-data.json:90]: @wolf99 if you're okay with this, can you let us know? I'll hold-off on merging until you say so.

**Comment by SleeplessByte** [comment]
For the person merging, see https://github.com/exercism/problem-specifications/pull/1767#discussion_r568778682 first.

File: exercises/high-scores/canonical-data.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (wolf99) [reply exercises/high-scores/canonical-data.json:90]: Yah, if that's the intended use, then this change LGTM

**Pull request merged**